splight_lib/client/datalake/remote_client.py

```diff
@@ -62,84 +65,104 @@ async def async_save(
+        url = self._base_url / f"{self._PREFIX}/write"
+        data = {
+            "collection": collection,
+            "records": instances,
+        }
+        response = await self._restclient.async_post(url, json=data)
         response.raise_for_status()
         return instances
 
     @retry(SPLIGHT_REQUEST_EXCEPTIONS, tries=3, delay=2, jitter=1)
-    def _raw_get(
+    def _get(
         self,
-        resource_name: str,
-        collection: str,
-        limit_: int = 50,
-        skip_: int = 0,
-        sort: Union[List, str] = ["timestamp__desc"],
-        group_id: Optional[Union[List, str]] = None,
-        group_fields: Optional[Union[List, str]] = None,
-        tzinfo: timezone = timezone(timedelta()),
+        asset: str,
+        attribute: str,
+        collection: str = "default",
+        sort_field: str = "timestamp",
+        sort_direction: int = -1,
+        limit: int = 10000,
         **filters,
-    ) -> List[Dict]:
-        # GET /datalake/data/
-        url = self._base_url / f"{self._PREFIX}/data/"
-
-        filters.update(
-            {
-                "source": collection,
-                "output_format": resource_name,
-                "sort": sort,
-                "limit_": limit_,
-                "skip_": skip_,
-                "group_id": group_id,
-                "group_fields": group_fields,
-                # "tzinfo": tzinfo
-            }
+    ):
+        url = self._base_url / f"{self._PREFIX}/read"
+        data_request = DataRequest(
+            collection=collection,
+            sort_field=sort_field,
+            sort_direction=sort_direction,
+            limit=limit,
+            traces=[
+                {
+                    "ref_id": "output",
+                    "type": "QUERY",
+                    "pipeline": [
+                        {
+                            "$match": {
+                                "asset": asset,
+                                "attribute": attribute,
+                            }
+                        }
+                    ],
+                    "expression": None,
+                }
+            ],
+            **filters,
         )
+        response = self._restclient.post(url, json=data_request.dict())
+        response.raise_for_status()
+        return response.json()
 
-        params = self._parse_params(**filters)
-        response = self._restclient.get(url, params=params)
+    @retry(SPLIGHT_REQUEST_EXCEPTIONS, tries=3, delay=2, jitter=1)
+    def raw_get(
+        self,
+        data_request: DataRequest,
+    ) -> List[Dict]:
+        url = self._base_url / f"{self._PREFIX}/read"
+
+        response = self._restclient.post(url, json=data_request.dict())
         response.raise_for_status()
-        output = response.json()["results"]
+        output = response.json()
 
         return output
 
     @retry(SPLIGHT_REQUEST_EXCEPTIONS, tries=3, delay=2, jitter=1)
-    async def _async_raw_get(
+    async def _async_get(
```

Comment:
Why we need a sync and async method? IMO always is better to have one interface to fetch one to write (agnostic of implementation)